src/sim/VacancyStore.hh

```diff
@@ -0,0 +1,48 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file VacancyStore.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "base/DeviceVector.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Manage device data for vacancies.
+ */
+class VacancyStore
```

Comment:
Do you anticipate the `VacancyStore` being anything but a detail-level class? For helper data like this, it might be easier for the helper class that calls the `InitializeTracks` functions to just store the `DeviceVector` internally. Your call of course.

Also, since I see the "shrinkable device vector" paradigm is used more than once here, it might be worth moving into the DeviceVector class, although the caveat is that it should only be used for trivially destructible objects.

Reply:
No, this was more of a placeholder since I didn't have the class to call those functions and manage that data yet, but ultimately storing it in that helper class makes the most sense.